src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java

```diff
@@ -0,0 +1,375 @@
+package ru.yandex.clickhouse.util;
+
+import com.google.common.io.LittleEndianDataInputStream;
+import com.google.common.primitives.UnsignedLong;
+import ru.yandex.clickhouse.settings.ClickHouseProperties;
+import ru.yandex.clickhouse.util.guava.StreamUtils;
+
+import java.io.Closeable;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.TimeZone;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static ru.yandex.clickhouse.util.ClickHouseRowBinaryStream.MILLIS_IN_DAY;
+
+public class ClickHouseRowBinaryInputStream implements Closeable {
+	private final LittleEndianDataInputStream in;
+	private final TimeZone timeZone;
+
+	public ClickHouseRowBinaryInputStream(InputStream is, TimeZone timeZone, ClickHouseProperties properties) {
+		this.in = new LittleEndianDataInputStream(is);
+		if (properties.isUseServerTimeZoneForDates()) {
+			this.timeZone = timeZone;
+		} else {
+			this.timeZone = TimeZone.getDefault();
+		}
+	}
+
+	public int readUnsignedLeb128() throws IOException {
+		int value = 0;
+		int read;
+		do {
+			read = in.read();
+			if (read == -1)
+				throw new EOFException();
+
+			value = (value << 7) | (read & 0x7f);
+		} while ((read & 0x80) != 0);
+
+		return value;
+	}
+
+	public void readBytes(byte[] bytes) throws IOException {
+		readBytes(bytes, 0, bytes.length);
+	}
+
+	public void readBytes(byte[] bytes, int offset, int length) throws IOException {
+		while (length > 0) {
+			int read = in.read(bytes, offset, length);
+			if (read == -1)
+				throw new EOFException();
+			offset += read;
+			length -= read;
+		}
+	}
+
+	public int readByte() throws IOException {
+		return in.read();
+	}
+
+	public boolean isNull() throws IOException {
+		int value = readByte();
+		if (value == -1)
+			throw new EOFException();
+
+		validateInt(value, 0, 1, "nullable");
+		return value != 0;
+	}
+
+	public String readString() throws IOException {
+		int length = readUnsignedLeb128();
+		byte[] bytes = new byte[length];
+		readBytes(bytes);
+
+		return new String(bytes, StreamUtils.UTF_8);
+	}
+
+	private void validateInt(int value, int minValue, int maxValue, String dataType) {
+		if (value < minValue || value > maxValue) {
+			throw new IllegalStateException("Not a " + dataType + " value: " + value);
+		}
+	}
+
+	public boolean readBoolean() throws IOException {
+		int value = readUInt8();
+		validateInt(value, 0, 1, "boolean");
+		return value != 0;
+	}
+
+	public short readUInt8() throws IOException {
+		return (short) in.readUnsignedByte();
+	}
+
+	/**
```

Comment:
I see the build had failed because of these javadoc-like comments. Documentation plugin expected javadoc to have specific fields. These should be simple comments or must be written with regards to javadoc rules.

Here is the build log with errors
https://api.travis-ci.org/v3/job/466062930/log.txt